code/modules/fish/fishtank.dm

```diff
@@ -0,0 +1,701 @@
+
+/obj/machinery/fishtank/attack_slime(mob/living/user as mob)
+	var/mob/living/carbon/slime/S = user
+	if (!S.is_adult)
+		return
+	attack_generic(user, rand(10, 15))
+
+/obj/machinery/fishtank/attack_hand(mob/user as mob)
+	if(HULK in user.mutations)
+		user.say(pick(";RAAAAAAAARGH!", ";HNNNNNNNNNGGGGGGH!", ";GWAAAAAAAARRRHHH!", "NNNNNNNNGGGGGGGGHH!", ";AAAAAAARRRGH!"))
+		user.visible_message("<span class='danger'>[user] smashes through [src]!</span>")
+		destroy()
+	else if (usr.a_intent == "harm")
+		user.changeNext_move(CLICK_CD_MELEE)
+		playsound(get_turf(src), 'sound/effects/glassknock.ogg', 80, 1)
+		usr.visible_message("\red [usr.name] bangs against the [src.name]!", \
+							"\red You bang against the [src.name]!", \
+							"You hear a banging sound.")
+	else
+		user.changeNext_move(CLICK_CD_MELEE)
+		playsound(src.loc, 'sound/effects/glassknock.ogg', 80, 1)
+		usr.visible_message("[usr.name] taps on the [src.name].", \
+							"You tap on the [src.name].", \
+							"You hear a knocking sound.")
+	return
+
+/obj/machinery/fishtank/proc/hit(var/damage, var/sound_effect = 1)
+	cur_health = max(0, cur_health - damage)
+	if(sound_effect)
+		playsound(loc, 'sound/effects/Glasshit.ogg', 75, 1)
+	check_health()
+
+/obj/machinery/fishtank/proc/attack_generic(mob/living/user as mob, damage = 0)	//used by attack_alien, attack_animal, and attack_slime
+	user.changeNext_move(CLICK_CD_MELEE)
+	user.do_attack_animation(src)
+	cur_health -= damage
+	if(cur_health <= 0)
+		user.visible_message("<span class='danger'>[user] smashes through \the [src]!</span>")
+		destroy()
+	else	//for nicer text~
+		user.visible_message("<span class='danger'>[user] smashes into \the [src]!</span>")
+		playsound(loc, 'sound/effects/Glasshit.ogg', 100, 1)
+		check_health()
+
+/obj/machinery/fishtank/attackby(var/obj/item/O, var/mob/user as mob)
+	//Welders repair damaged tanks on help intent, damage on all others
+	if(istype(O, /obj/item/weapon/weldingtool))
+		var/obj/item/weapon/weldingtool/W = O
+		if(user.a_intent == "help")
+			if(W.isOn())
+				if(cur_health < max_health)
+					usr << "You repair some of the cracks on \the [src]."
+					cur_health += 20
+					check_health()
+				else
+					usr << "There is no damage to fix!"
+			else
+				if(cur_health < max_health)
+					usr << "[W.name] must on to repair this damage."
+		else
+			user.changeNext_move(CLICK_CD_MELEE)
+			hit(W.force)
+		return
+	//Open reagent containers add and remove water
+	if(O.is_open_container())
+		if(istype(O, /obj/item/weapon/reagent_containers/glass))
+			if(lid_switch)
+				usr << "Open the lid on [src] first!"
+				return
+			var/obj/item/weapon/reagent_containers/glass/C = O
+			//Containers with any reagents will get dumped in
+			if(C.reagents.total_volume)
+				var/water_value = 0
+				water_value += C.reagents.get_reagent_amount("water")				//Water is full value
+				water_value += C.reagents.get_reagent_amount("holywater")			//Holywater is full value
```

Comment:
Oh, definitely, I was thinking like 1.1; just barely enough to cause an eyebrow raise.